src/alert_processing/file_system_status.py

```diff
@@ -0,0 +1,65 @@
+import logging
+from datetime import datetime, timedelta, timezone
+from pathlib import Path
+from typing import Iterable
+
+import pandas as pd
+
+__all__ = [
+    "get_modified_time",
+    "check_update_time",
+]
+
+logger = logging.getLogger(__name__)
+
+def get_modified_time(
+        files: Iterable[Path],
+        skip_dir: bool = False,
+        skip_hidden: bool = True,
+) -> pd.DataFrame:
+    paths = []
+    for path in files:
+        if skip_dir and path.is_dir():
+            continue
+        if skip_hidden and path.name[0] == '.':
+            continue
+        file_info = dict(
+            stem=path.stem,
+            path=path.as_posix(),
+            modified_datetime=datetime.fromtimestamp(
+                path.stat().st_mtime,
```

Comment:
I prefer using `Path` instances from `pathlib` instead of working on raw string.
Then I have properties like `file_path.stem` and `file_path.name` and can use a fluent api style methods when processing my paths. `file_path.parent.joinpath('sibling.txt')`.